### PR TITLE
Make C-enums into idiomatic rust enums

### PR DESCRIPTION
**Pull request by CGMossa**
There are several "inferred" enums in the R-API. This setting ensures that
they become idiomatic Rust enums.
There are still missing ones, namely the macro-defined-enums, but maybe I'll add those somewhere else.

### CODE REVIEW

File: bindings/bindings-linux-aarch64-R4.2.rs [hunk 1]
```diff
@@ -284,11 +284,14 @@ pub struct _IO_FILE {
     pub _mode: ::std::os::raw::c_int,
     pub _unused2: [::std::os::raw::c_char; 20usize],
 }
-#[doc = ", MAYBE"]
```
Comment (Ilia-Kosenkov) [inline bindings/bindings-linux-aarch64-R4.2.rs:287]: This is some weird comment likely inferred from the header file. Can we drop/override it specifically for this one enum?

File: bindings/bindings-linux-aarch64-R4.2.rs [hunk 2]
```diff
@@ -348,28 +354,37 @@ pub struct R_allocator {
...
-pub const R_pstream_format_t_R_pstream_any_format: R_pstream_format_t = 0;
-pub const R_pstream_format_t_R_pstream_ascii_format: R_pstream_format_t = 1;
-pub const R_pstream_format_t_R_pstream_binary_format: R_pstream_format_t = 2;
-pub const R_pstream_format_t_R_pstream_xdr_format: R_pstream_format_t = 3;
-pub const R_pstream_format_t_R_pstream_asciihex_format: R_pstream_format_t = 4;
-pub type R_pstream_format_t = u32;
+#[repr(u32)]
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
+pub enum R_pstream_format_t {
+    R_pstream_any_format = 0,
```
Comment (Ilia-Kosenkov) [inline bindings/bindings-linux-aarch64-R4.2.rs:382]: Crazy how there are like 3 enums within one screen and they all have different naming convention. Ideally I'd prefer to drop `R_pstream` prefix here

File: bindings/bindings-linux-aarch64-R4.2.rs [hunk 3]
```diff
@@ -624,42 +645,54 @@ pub type R_altstring_Is_sorted_method_t =
...
-pub const R_GE_lineend_GE_BUTT_CAP: R_GE_lineend = 2;
-pub const R_GE_lineend_GE_SQUARE_CAP: R_GE_lineend = 3;
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
+pub enum GEUnit {
+    #[doc = "native device coordinates (rasters)"]
+    GE_DEVICE = 0,
+    #[doc = "normalised device coordinates x=(0,1), y=(0,1)"]
+    GE_NDC = 1,
+    GE_INCHES = 2,
+    GE_CM = 3,
```
Comment (Ilia-Kosenkov) [inline bindings/bindings-linux-aarch64-R4.2.rs:657]: <s>I think we lost docstrings here, no?</s> My bad, did not notice rearrangement.

File: bindings/bindings-linux-aarch64-R4.2.rs (hunk 1 above)
Comment (CGMossa) [reply bindings/bindings-linux-aarch64-R4.2.rs:287]: Unfortunately, I cannot. these are not modifiable. Eventually, I'd figure out a good system to snap the comments, but it is just difficult with the chaos that is R-API's comments..

File: bindings/bindings-linux-aarch64-R4.2.rs (hunk 2 above)
Comment (CGMossa) [reply bindings/bindings-linux-aarch64-R4.2.rs:382]: I don't think I can

```c

typedef enum {
    R_pstream_any_format,
    R_pstream_ascii_format,
    R_pstream_binary_format,
    R_pstream_xdr_format,
    R_pstream_asciihex_format
} R_pstream_format_t;
```
They wrote them like that :(

**Comment by Ilia-Kosenkov** [comment]
Hm, should we already bump `libR-sys` version to some `0.7.0-dev` value? This is a significant breaking change, I want to see a difference between the released `0.6.0` and what is currently developed.

**Comment by CGMossa** [comment]
Sure. Please make a PR that bumps the version.

**Comment by Ilia-Kosenkov** [comment]
Checking my comments, I do not get it why can't we mark this as `0.7.0`?

**Comment by CGMossa** [comment]
Bumped version.

**Comment by CGMossa** [comment]
> Checking my comments, I do not get it why can't we mark this as `0.7.0`?

Hard to remember.. I guess I saw that you'd make separate PRs to bump the version in rextendr. And I asked you to do it here maybe because I thought it was more elaborate than just changing a file.

**Review by Ilia-Kosenkov — APPROVED** [review]

**Pull request merged**